akka-stream-tests/src/test/scala/akka/stream/scaladsl/RestartSpec.scala

```diff
@@ -667,6 +675,65 @@ class RestartSpec extends StreamSpec(Map("akka.test.single-expect-default" -> "1
 
       created.get() should ===(2)
     }
-  }
 
+    // onlyOnFailures -->
+    "stop on cancellation when using onlyOnFailuresWithBackoff" in {
+      val onlyOnFailures = true
+      val (created, source, flowInProbe, flowOutProbe, sink) = setupFlow(shortMinBackoff, shortMaxBackoff, -1, onlyOnFailures)
+
+      source.sendNext("a")
+      flowInProbe.requestNext("a")
+      flowOutProbe.sendNext("b")
+      sink.requestNext("b")
+
+      source.sendNext("cancel")
+      // This will complete the flow in probe and cancel the flow out probe
+      flowInProbe.request(2)
+      flowInProbe.expectNext("in complete")
+
+      source.expectCancellation()
+
+      created.get() should ===(1)
+    }
+
+    "stop on completion when using onlyOnFailuresWithBackoff" in {
+      val onlyOnFailures = true
+      val (created, source, flowInProbe, flowOutProbe, sink) = setupFlow(shortMinBackoff, shortMaxBackoff, -1, onlyOnFailures)
+
+      source.sendNext("a")
+      flowInProbe.requestNext("a")
+      flowOutProbe.sendNext("b")
+      sink.requestNext("b")
+
+      flowOutProbe.sendNext("complete")
+      sink.request(1)
+      sink.expectComplete()
```

Comment:
request should technically not be needed there,

Reply:
thks for the tip